source/agora/utils/TxBuilder.d

```diff
@@ -646,21 +807,39 @@ unittest
 /// Test with unfrozen remainder
 unittest
 {
-    const result = TxBuilder(GenesisBlock.payments.front)
-        .draw(Amount.UnitPerCoin * 50_000, WK.Keys.byRange.map!(k => k.address).take(3))
+    const freezeAmount = Amount(50_000L * 10_000_000L);
+    const tx = TxBuilder(GenesisBlock.payments.front)
+        .draw(freezeAmount, WK.Keys.byRange.map!(k => k.address).takeExactly(1))
         .sign(OutputType.Freeze);
 
-    // This transaction has 4 outputs (3 freeze + 1 refund)
-    assert(result.inputs.length == 8);
-    assert(result.outputs.length == 4);
+    // This transaction has 2 outputs (1 freeze + 1 refund)
+    assert(tx.inputs.length == 8);
+    assert(tx.outputs.length == 2);
 
-    // 488M / 3
-    assert(result.outputs == [
-        Output(Amount(50_000_0000_000L), WK.Keys.A.address, OutputType.Freeze),
-        Output(Amount(50_000_0000_000L), WK.Keys.C.address, OutputType.Freeze),
-        Output(Amount(50_000_0000_000L), WK.Keys.D.address, OutputType.Freeze),
-        Output(Amount(487_850_000_0000_000L), WK.Keys.Genesis.address),
-    ].sort.array);
+    auto fees = fee_rate * tx.sizeInBytes;
+    assert(tx.outputs.count!(o => o.value == freezeAmount && o.type == OutputType.Freeze) == 1);
+    auto refund = sumOfGenesisFirstTxOutputs() - freezeAmount - fees;
+    assert(tx.outputs.count!(o => o.value == refund && o.type == OutputType.Payment) == 1);
```

Comment:
Doesn't this essentially revert the diff introduced in the previous commit ?

Reply:
Well spotted, I'll revert it.